react/features/recent-list/middleware.js

```diff
@@ -0,0 +1,111 @@
+/* @flow */
+
+import { LIST_SIZE } from './constants';
+import { getRecentUrls, updaterecentUrls } from './functions';
+
+import { SET_ROOM, CONFERENCE_WILL_LEAVE } from '../base/conference';
+import { MiddlewareRegistry } from '../base/redux';
+
+/**
+ * Middleware that captures joined rooms so then it can be saved to
+ * {@code localStorage}
+ *
+ * @param {Store} store - Redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case CONFERENCE_WILL_LEAVE:
+        return _updateConferenceDuration(store, next, action);
+
+    case SET_ROOM:
+        return _storeJoinedRoom(store, next, action);
+    }
+
+    return next(action);
+});
+
+/**
+* Stores the recently joined room in {@code localStorage}.
+*
+* @param {Store} store - The redux store in which the specified action is being
+* dispatched.
+* @param {Dispatch} next - The redux dispatch function to dispatch the
+* specified action to the specified store.
+* @param {Action} action - The redux action CONFERENCE_JOINED which is being
+* dispatched in the specified store.
+* @returns {Object} The new state that is the result of the reduction of the
+* specified action.
+*/
+function _storeJoinedRoom(store, next, action) {
+    const result = next(action);
+    const { room } = action;
+
+    if (room) {
+        const { locationURL } = store.getState()['features/base/connection'];
+        const conferenceLink = locationURL.href;
+        let recentUrls = getRecentUrls();
```

Comment:
you can make this a const if you combine it with the filer

Reply:
Fixed, thanks!